podcasts/PaidFeature.swift

```diff
@@ -0,0 +1,84 @@
+import Combine
+import PocketCastsServer
+import SwiftUI
+
+// MARK: - Features
+
+/// To add a new feature:
+///     1. Add a new static var for the feature name and tier it should be unlocked with
+///
+///     Template:
+///         static var <#FeatureName#>: PaidFeature = .init(tier: <#Tier#>)
+///
+///     2. Check the unlock state using `PaidFeature.hello.isUnlocked`
+///
+extension PaidFeature {
+    static var bookmarks: PaidFeature = .init(tier: .patron)
+}
+
+/// A `PaidFeature` represents a feature that is unlocked with a subscription tier, and is considered to be unlocked if the tier
+/// is equal to or higher than the set `tier` value.
+///
+/// The unlock state is self managed by listening for relevant notifications and updating the `isUnlocked` property accordingly.
+///
+/// Since this is a subclass of `ObservableObject` it can easily be used in SwiftUI. Outside SwiftUI, you can use the `objectWillChange`
+/// publisher to be notified about changes.
+///
+/// And while the class is an `ObservableObject` it doesn't use any `@Published` properties and instead manually triggers `objectWillChange`.
+/// This is done to ensure future compatibility of any property changes by not allowing listeners to directly access a published properties publisher.
+///
+class PaidFeature: ObservableObject {
+    /// Whether the feature is unlocked for the active subscription tier
+    var isUnlocked: Bool {
+        subscriptionHelper.activeTier >= tier
```

Comment:
Since all we use here is `activeTier` do we really need to inject `SubscriptionHelper` here? Can't we just inject the `activeTier` on `init` and then update it when the listeners are triggered?

Reply:
Hmm. I don't think that would work because if we injected the tier we would still need to access the SubscriptionHelper in order to update the value.

Reply:
Got it!

Another suggestion (feel free to tackle in the future if you'd prefer and, of course, if you agree): what do you think about creating a `SubscriptionHelper` singleton and injecting that? This singleton would contain only the `activeTier` (which accesses the static `activeTier` property) but I think a singleton is better than a whole static class. Of course, you would make the change only here because changing other places that use `SubscriptionHelper` is out of scope.

Reply:
I like that idea, it makes more sense and pushes us in a better directly. I updated the code here:
- 67840024b07cd5cba2dcafd1edc21b6ab3853455
- 9c8a47344c73c2af5911a234d8a93f478793a955